### PR TITLE
Export timelines for host

### PR DESCRIPTION
**Pull request by TamazC**
## Description

** This PR enables to export timelines for given host 

**Fixes** # MON-13542

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Timelines for host could be downloaded at new endpoint :
`http://{http_host}/centreon/api/latest/monitoring/hosts/${host_id}/timeline/download`

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by dmyios — DISMISSED** [review]

File: src/Centreon/Application/Controller/Monitoring/TimelineController.php [hunk 1]
```diff
@@ -129,22 +115,45 @@ public function getServiceTimeline(
...
         return $this->view(
             [
-                'result' => $this->getTimelines($hostId, $serviceId),
+                'result' => $this->getTimelinesByHostIdAndServiceId($hostId, $serviceId),
                 'meta' => $requestParameters->toArray()
             ]
         )->setContext($context);
     }
 
+    public function downloadHostTimeline(int $hostId, RequestParametersInterface $requestParameters): StreamedResponse
```
Comment (jeremyjaouen) [inline src/Centreon/Application/Controller/Monitoring/TimelineController.php:129]: phpdoc

File: src/Centreon/Application/Controller/Monitoring/TimelineController.php [hunk 2]
```diff
@@ -129,22 +115,45 @@ public function getServiceTimeline(
...
         RequestParametersInterface $requestParameters
     ): StreamedResponse {
         $this->denyAccessUnlessGrantedForApiRealtime();
+        $this->addDownloadParametersInRequestParameters($requestParameters);
+        $timeLines = $this->formatTimeLinesForDownload($this->getTimelinesByHostIdAndServiceId($hostId, $serviceId));
+
+        return $this->streamTimeLines($timeLines);
+    }
+
+    private function addDownloadParametersInRequestParameters(RequestParametersInterface $requestParameters): void
```
Comment (jeremyjaouen) [inline src/Centreon/Application/Controller/Monitoring/TimelineController.php:161]: phpdoc

File: src/Centreon/Application/Controller/Monitoring/TimelineController.php [hunk 3]
```diff
@@ -129,22 +115,45 @@ public function getServiceTimeline(
...
+    public function downloadHostTimeline(int $hostId, RequestParametersInterface $requestParameters): StreamedResponse
+    {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+        $this->addDownloadParametersInRequestParameters($requestParameters);
+        $timeLines = $this->formatTimeLinesForDownload($this->getTimelinesByHostId($hostId));
+
+        return $this->streamTimeLines($timeLines);
+    }
+
     public function downloadServiceTimeline(
```
Comment (jeremyjaouen) [inline src/Centreon/Application/Controller/Monitoring/TimelineController.php:145]: phpdoc

File: src/Centreon/Application/Controller/Monitoring/TimelineController.php [hunk 4]
```diff
@@ -214,55 +207,88 @@ public function getMetaServiceTimeline(
...
         return $this->view(
             [
-                'result' => $timeline,
+                'result' => $this->getMetaServiceTimelineById($metaId),
                 'meta' => $requestParameters->toArray()
             ]
         )->setContext($context);
     }
 
+    public function downloadMetaserviceTimeline(
```
Comment (jeremyjaouen) [inline src/Centreon/Application/Controller/Monitoring/TimelineController.php:259]: phpdoc

File: src/Centreon/Application/Controller/Monitoring/TimelineController.php [hunk 5]
```diff
@@ -214,55 +207,88 @@ public function getMetaServiceTimeline(
...
+        if ($service === null) {
+            $errorMsg = sprintf(_('Service %d on host %d not found'), $hostId, $serviceId);
+            throw new EntityNotFoundException($errorMsg);
+        }
+        $service->setHost($host);
+
+        return $this->timelineService->findTimelineEventsByService($service);
+    }
+
+    private function getHostById(int $hostId): Host
```
Comment (jeremyjaouen) [inline src/Centreon/Application/Controller/Monitoring/TimelineController.php:307]: phpdoc

**Review by jeremyjaouen — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**